code/modules/antagonists/changeling/abomination.dm

```diff
@@ -0,0 +1,102 @@
+/*
+	- CHANGELING ABOMINATION -
+
+	Contents:
+		Reagent
+		Monster
+*/
+
+/datum/reagent/shiftium
+	name = "shiftium"
+	description = "A strange stimulant."
+	color = "#9D5A99"
+	overdose_threshold = 10
+	taste_description = "savory with a bit of blood"
+	metabolization_rate = 0.3 * REAGENTS_METABOLISM
+	var/obj/shapeshift_holder/shapeshiftdata
+
+/datum/reagent/shiftium/on_mob_metabolize(mob/living/L)
+	L.adjustStaminaLoss(-5, 0)
+	..()
+
+/datum/reagent/shiftium/on_mob_end_metabolize(mob/living/L)
+	..()
+	if(!shapeshiftdata)
+		return
+	shapeshiftdata.restore()
+	REMOVE_TRAIT(L, CHANGELING_HIVEMIND_MUTE, type)
+
+/datum/reagent/shiftium/overdose_start(mob/living/L)
+	if (L.mind?.has_antag_datum(/datum/antagonist/changeling))
+		to_chat(L,"<span class='danger'>You struggle to maintain your form!</span>")
+	..()
+
+/datum/reagent/shiftium/overdose_process(mob/living/L)
+	L.adjustStaminaLoss(5, 0)
+	if (prob(volume/2))
+		var/datum/antagonist/changeling/changeling = L.mind?.has_antag_datum(/datum/antagonist/changeling)
+		if(changeling)
+			metabolization_rate = 10 * REAGENTS_METABOLISM
+			ADD_TRAIT(L, CHANGELING_HIVEMIND_MUTE, type)
+
+			playsound(L, 'sound/magic/demon_consume.ogg', 30, 1)
+			L.visible_message("<span class='warning'>[L]'s body uncontrolably transforms into an abomination!</span>", "<span class='boldwarning'>Your body uncontrolably transforms, revealing your true form!</span>")
+
+			polymorph_target(L,volume/overdose_threshold)
+	..()
+
+/datum/reagent/shiftium/proc/polymorph_target(mob/living/L, var/dur)
+	if(L.anti_magic_check())
+		return
+	shapeshiftdata = locate() in L
```

Comment:
🤔

Reply:
I am pretty sure that you have to check for `L`'s `loc` instead of `contents`; `shapeshift_holder` has the original shape as its content and itself as new shape's content: new shape > shapeshifter holder > original shape